### PR TITLE
Add /timestamp

### PR DESCRIPTION
**Pull request by mctaylors**
Original idea from @Octol1ttle

### CODE REVIEW

**Comment by mctaylors** [comment]
@Octol1ttle, looks like pull-request-size bot is malfunctioning...

File: src/Commands/ToolsCommandGroup.cs [hunk 1]
```diff
@@ -289,4 +289,74 @@ private async Task<Result> SendRandomNumberAsync(int max, int min, IUser user, I
 
         return await _feedback.SendContextualEmbedResultAsync(embed, ct);
     }
+
+    /// <summary>
+    ///     A slash command that shows current or counted timestamp.
+    /// </summary>
+    /// <param name="offset">The offset to count timestamp.</param>
```
Comment (Octol1ttle) [inline src/Commands/ToolsCommandGroup.cs:296]:
```suggestion
    ///     A slash command that shows the current timestamp with an optional offset in all styles supported by Discord.
    /// </summary>
    /// <param name="offset">The offset for the current timestamp.</param>
```

File: src/Commands/ToolsCommandGroup.cs [hunk 2]
```diff
@@ -289,4 +289,74 @@ private async Task<Result> SendRandomNumberAsync(int max, int min, IUser user, I
...
+    /// <returns>
+    ///     A feedback sending result which may or may not have succeeded.
+    /// </returns>
+    [Command("timestamp")]
+    [DiscordDefaultDMPermission(false)]
+    [Description("Gets timestamp")]
```
Comment (Octol1ttle) [inline src/Commands/ToolsCommandGroup.cs:302]:
```suggestion
    [Description("Shows a timestamp in all styles")]
```

File: src/Commands/ToolsCommandGroup.cs [hunk 3]
```diff
@@ -289,4 +289,74 @@ private async Task<Result> SendRandomNumberAsync(int max, int min, IUser user, I
...
+            return Result.FromError(userResult);
+        }
+
+        var data = await _guildData.GetData(guildId, CancellationToken);
+        Messages.Culture = GuildSettings.Language.Get(data.Settings);
+
+        return await SendRandomNumberAsync(offset, user, CancellationToken);
+    }
+
+    private async Task<Result> SendRandomNumberAsync(TimeSpan? offset, IUser user, CancellationToken ct)
```
Comment (Octol1ttle) [inline src/Commands/ToolsCommandGroup.cs:325]: `SendRandomNumberAsync`?

File: src/Commands/ToolsCommandGroup.cs [hunk 4]
```diff
@@ -289,4 +289,74 @@ private async Task<Result> SendRandomNumberAsync(int max, int min, IUser user, I
...
+            TimestampStyle.ShortDateTime,
+            TimestampStyle.LongDateTime,
+            TimestampStyle.RelativeTime
+        };
+
+        foreach (var style in allStyles)
+        {
+            description.Append("- ").Append(Markdown.InlineCode(Markdown.Timestamp(timestamp, style)))
+                .Append(" → ").AppendLine(Markdown.Timestamp(timestamp, style));
+        }
```
Comment (Octol1ttle) [inline src/Commands/ToolsCommandGroup.cs:352]: Put result of `Markdown.Timestamp` in a variable (put text cursor on the method call and press Ctrl+Alt+V in Rider)

**Review by Octol1ttle — CHANGES_REQUESTED** [review]

File: src/Commands/ToolsCommandGroup.cs [hunk 5]
```diff
@@ -289,4 +289,74 @@ private async Task<Result> SendRandomNumberAsync(int max, int min, IUser user, I
...
+    private static readonly TimestampStyle[] AllStyles =
+    {
+        TimestampStyle.ShortDate,
+        TimestampStyle.LongDate,
+        TimestampStyle.ShortTime,
+        TimestampStyle.LongTime,
+        TimestampStyle.ShortDateTime,
+        TimestampStyle.LongDateTime,
+        TimestampStyle.RelativeTime
+    };
```
Comment (Octol1ttle) [inline src/Commands/ToolsCommandGroup.cs:334]: Put this above `ExecuteTimestampAsync`

File: src/Commands/ToolsCommandGroup.cs [hunk 6]
```diff
@@ -289,4 +289,74 @@ private async Task<Result> SendRandomNumberAsync(int max, int min, IUser user, I
...
+    /// <param name="offset">The offset for the current timestamp.</param>
+    /// <returns>
+    ///     A feedback sending result which may or may not have succeeded.
+    /// </returns>
+    [Command("timestamp")]
+    [DiscordDefaultDMPermission(false)]
+    [Description("Shows a timestamp in all styles")]
+    [UsedImplicitly]
+    public async Task<Result> ExecuteTimestampAsync(
+        [Description("Add an offset from now")]
```
Comment (Octol1ttle) [inline src/Commands/ToolsCommandGroup.cs:305]:
```suggestion
        [Description("Offset from current time")]
```

**Review by Octol1ttle — APPROVED** [review]

**Pull request merged**